dev/com.ibm.ws.security.common/test/com/ibm/ws/security/common/web/WebUtilsTest.java

```diff
@@ -299,4 +315,521 @@ public void decodeCookie_fullEncodedSpecialChars() {
         }
     }
 
+    @Test
+    public void testStripSecretFromUrlNoChange() {
+        assertNull(WebUtils.stripSecretFromUrl(null, null));
+        assertNull(WebUtils.stripSecretFromUrl(null, "something"));
+
+        String input = "";
+        String expected = input;
+        String secret = "secret";
+        String value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "something";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, null);
+        assertEquals(expected, value);
+
+        input = "something";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, "");
+        assertEquals(expected, value);
+
+        input = "abc";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "something_longer_than_client_secret";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "&client_secret=";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "&CLIENT_SECRET=x";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "http://localhost";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "http://localhost?";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "http://localhost?client_secret";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+
+        input = "http://localhost?client_secret=";
+        expected = input;
+        value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testStripSecretFromUrl1() {
+        String input = "&client_secret=x";
+        String expected = "&client_secret=*****";
+        String secret = "client_secret";
+        String value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testStripSecretFromUrl2() {
+        String input = "abc&client_secret=password&&client_secret=password";
+        String expected = "abc&client_secret=*****&&client_secret=*****";
+        String secret = "client_secret";
+        String value = WebUtils.stripSecretFromUrl(input, secret);
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testStripSecretFromUrl3() {
```

Comment:
This is exactly the same as `testStripSecretFromUrl2`.

Reply:
oops, too many copies.  I'll switch them up a little.